sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,334 @@
+ * @param[out] out_mqtt_client_id The output #az_span containing the MQTT client id.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_id_get(
+    az_iot_pnp_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id);
+
+/*
+ *
+ * SAS Token APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Gets the Shared Access clear-text signature.
+ * @details The application must obtain a valid clear-text signature using this API, sign it using
+ *          HMAC-SHA256 using the Shared Access Key as password then Base64 encode the result.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] token_expiration_epoch_time The time, in seconds, from 1/1/1970.
+ * @param[in] signature An empty #az_span with sufficient capacity to hold the SAS signature.
+ * @param[out] out_signature The output #az_span containing the SAS signature.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_sas_signature_get(
+    az_iot_pnp_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
+    az_span* out_signature);
+
+/**
+ * @brief Gets the MQTT password.
+ * @note The MQTT password must be an empty string if X509 Client certificates are used. Use this
+ *       API only when authenticating with SAS tokens.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] base64_hmac_sha256_signature The Base64 encoded value of the HMAC-SHA256(signature,
+ *                                         SharedAccessKey). The signature is obtained by using
+ *                                         #az_iot_pnp_client_sas_signature_get.
+ * @param[in] key_name The Shared Access Key Name (Policy Name). This is optional. For security
+ *                     reasons we recommend using one key per device instead of using a global
+ *                     policy key.
+ * @param[in] mqtt_password An empty #az_span with sufficient capacity to hold the MQTT password.
+ * @param[out] out_mqtt_password The output #az_span containing the MQTT password.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_sas_password_get(
+    az_iot_pnp_client const* client,
+    az_span base64_hmac_sha256_signature,
+    az_span key_name,
+    az_span mqtt_password,
+    az_span* out_mqtt_password);
+
+/*
+ *
+ * PnP Telemetry APIs
+ *
+ */
+
+/**
+ * @brief Azure IoT PnP telemetry options.
+ *
+ */
+typedef struct az_iot_pnp_client_telemetry_options
+{
+  struct
+  {
+    // OPEN: I can't find samples in Azure SDK modeling this.  Closest is
+    // sdk\storage\blobs\inc\az_storage_blobs.h where it has 'az_span option" in
+    // az_storage_blobs_blob_download_options that's ignored in practice.
```

Comment:
Yes, this is fine.

Reply:
Mooted because we're going to the pvReserved (uhm, reserved.  Still want to call it pvResorved per Windows/Hungarian days...)